runtime/jvmti/jvmtiCapability.c

```diff
@@ -373,7 +373,10 @@ jvmtiAddCapabilities(jvmtiEnv* env,
 
 #if JAVA_SPEC_VERSION >= 11
 		if (newCapabilities.can_generate_sampled_object_alloc_events) {
-			if (J9_ARE_ANY_BITS_SET(jvmtiData->flags, J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED)) {
+			if (J9_ARE_ANY_BITS_SET(jvmtiData->flags, J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED)
+				|| (J9_GC_POLICY_METRONOME == vm->gcPolicy)
+				|| (J9_GC_POLICY_BALANCED == vm->gcPolicy)
+			) {
```

Comment:
Should the `can_generate_sampled_object_alloc_events` event be removed from the potential set of capabilities when running with metronome or balanced? ie https://github.com/eclipse/openj9/blob/e363d8001fcfa8eef2641b326b194ec9503959c7/runtime/jvmti/jvmtiCapability.c#L178

Reply:
yeah, this is the right place. Thanks @DanHeidinga , PR is updated, please have another look.